app/questionnaire/path_finder.py

```diff
@@ -80,16 +94,30 @@ def _build_routing_path_block_ids(self, blocks, current_location):
 
         while block_index < len(blocks):
             block = blocks[block_index]
-
-            is_skipping = block.get("skip_conditions") and evaluate_skip_conditions(
-                block["skip_conditions"],
-                self.schema,
-                self.metadata,
-                self.answer_store,
-                self.list_store,
-                current_location=current_location,
-                routing_path_block_ids=routing_path_block_ids,
-            )
+            if isinstance(block.get("skip_conditions"), dict):
+                when_rule_evaluator = WhenRuleEvaluator(
+                    self.schema,
+                    self.answer_store,
+                    self.list_store,
+                    self.metadata,
+                    self.response_metadata,
+                    location=current_location,
+                    routing_path_block_ids=routing_path_block_ids,
+                )
+                when_rule = block["skip_conditions"].get("when")
+                is_skipping = block["skip_conditions"] and when_rule_evaluator.evaluate(
+                    when_rule
+                )
```

Comment:
Can be inline, since it has to existing at this point so `.get()` isn't needed. You also don't need the first check to see if you have skip conditions as you should at this point. Empty dict would be an invalid schema.

```suggestion
                is_skipping = when_rule_evaluator.evaluate(
                    skip_conditions["when"]
                )
```